src/modules/claim/resolvers/credential.resolver.ts

```diff
@@ -76,7 +76,11 @@ export class RoleCredentialResolver implements CredentialResolver {
     return tokens
       .map((token) => {
         try {
-          return jwt.decode(token) as RolePayload;
+          const decoded = jwt.decode(token) as RolePayload;
+          return {
+            eip191Jwt: token,
```

Comment:
Are you sure that `token` should be RolePayload? I thought we are going to store VC/EIP191 on IPFS. If that is right then error is casting decoded token to `RolePayload`

Reply:
@JGiter further down in this method (on line 88), there is a `.filter(isEIP191Jwt)` which is looking for both the `eip191Jwt` property and `payload.claimData.claimType`. https://github.com/energywebfoundation/ew-credentials/blob/861feac6593b4807f1fc72e9f184357f7ac007fd/packages/vc-verification/src/utils/formatters.ts#L98
So it makes sense to me that this would fix the issue.

Reply:
Hmm I think so. This is the return from the decode and below that the RolePayload interface
{
 claimData: {
  requestorFields: [],
  claimType: ‘errortestaug1.roles.suborgs.whitney.iam.ewc’,
  claimTypeVersion: 1,
  issuerFields: []
 },
 did: ‘did:ethr:volta:0x17b65C8C9746F87c82cc6f7C4FC38fCA5f1AeB75’,
 signer: ‘did:ethr:volta:0x17b65C8C9746F87c82cc6f7C4FC38fCA5f1AeB75’,
 credentialStatus: {
  id: ‘https://identitycache-dev.energyweb.org/v1/status-list/urn:uuid:5829660e-46b1-4ee5-8072-0db6e67ef7e7’,
  type: ‘StatusList2021Entry’,
  statusPurpose: ‘revocation’,
  statusListIndex: ‘0’,
  statusListCredential: ‘https://identitycache-dev.energyweb.org/v1/status-list/urn:uuid:5829660e-46b1-4ee5-8072-0db6e67ef7e7’
 },
 iss: ‘did:ethr:volta:0x17b65C8C9746F87c82cc6f7C4FC38fCA5f1AeB75’,
 sub: ‘did:ethr:volta:0x17b65C8C9746F87c82cc6f7C4FC38fCA5f1AeB75’,
 exp: 1659462891
}

export interface RolePayload extends JwtPayload {
    credentialStatus?: StatusList2021Entry;
    claimData: ClaimData;
    signer: string;
}

And then that object is used here:
```
 this.serviceEndpointsToEIP191(resolvedEndpoints).find(
        (token) => token.payload.claimData.claimType === namespace
      )
```
Whereas before it was just sending the decoded token, so it didn't have the key token on the object, so it wasn't finding a match.